endec/rubix.go

```diff
@@ -318,43 +221,25 @@ func decodeData(serialData *SerialData, header MetaDataKey, data interface{}) er
 	return nil
 }
 
-var serialMap = map[int]MetaData{
-	MDK_TEMP:             {FIXEDPOINT, -45, 120, 2, 0},
-	MDK_RH:               {FIXEDPOINT, 0, 100, 2, 0},
-	MDK_LUX:              {FIXEDPOINT, 0, 65534, 0, 0},
-	MDK_MOVEMENT:         {FIXEDPOINT, 0, 1, 0, 0},
-	MDK_COUNTER:          {FIXEDPOINT, 0, 1048576, 0, 0},
-	MDK_DIGITAL:          {FIXEDPOINT, 0, 1, 0, 0},
-	MDK_VOLTAGE_0_10:     {FIXEDPOINT, 0, 10, 2, 0},
-	MDK_MILLIAMPS_4_20:   {FIXEDPOINT, 4, 20, 2, 0},
-	MDK_OHM:              {FIXEDPOINT, 0, 1048576, 0, 0},
-	MDK_CO2:              {FIXEDPOINT, 0, 400, 0, 0},
-	MDK_BATTERY_VOLTAGE:  {FIXEDPOINT, 0, 6, 1, 0},
-	MDK_PUSH_FREQUENCY:   {FIXEDPOINT, 0, 2000, 0, 0},
-	MDK_RAW:              {FIXEDPOINT, 0, 1, 3, 0},
-	MDK_UO:               {FIXEDPOINT, 0, 1, 3, 0},
-	MDK_UI:               {FIXEDPOINT, 0, 1, 3, 0},
-	MDK_DO:               {FIXEDPOINT, 0, 1, 0, 0},
-	MDK_DI:               {FIXEDPOINT, 0, 1, 0, 0},
-	MDK_FIRMWARE_VERSION: {FIXEDPOINT, 0, 255, 0, 0},
-	MDK_HARDWARE_VERSION: {FIXEDPOINT, 0, 255, 0, 0},
-	MDK_UINT_8:           {DATAPOINT, 0, 0, 0, 1},
-	MDK_INT_8:            {DATAPOINT, 0, 0, 0, 1},
-	MDK_UINT_16:          {DATAPOINT, 0, 0, 0, 2},
-	MDK_INT_16:           {DATAPOINT, 0, 0, 0, 2},
-	MDK_UINT_32:          {DATAPOINT, 0, 0, 0, 4},
-	MDK_INT_32:           {DATAPOINT, 0, 0, 0, 4},
-	MDK_UINT_64:          {DATAPOINT, 0, 0, 0, 8},
-	MDK_INT_64:           {DATAPOINT, 0, 0, 0, 8},
-	MDK_BOOL:             {FIXEDPOINT, 0, 1, 0, 0},
-	MDK_CHAR:             {DATAPOINT, 0, 0, 0, 1},
-	MDK_FLOAT:            {DATAPOINT, 0, 0, 0, 4},
-	MDK_DOUBLE:           {DATAPOINT, 0, 0, 0, 8}}
-
 func generateFieldName(baseName string, hasPosition bool, pos *uint8) string {
 	if !hasPosition {
 		*pos++
 	}
+	if *pos == 1 {
+		return "Operation"
+	} else if *pos == 2 {
+		return "Mode"
+	} else if *pos == 3 {
+		return "Cool Temperature"
+	} else if *pos == 4 {
+		return "Heat Temperature"
+	} else if *pos == 5 {
+		return "Fan Speed"
+	} else if *pos == 6 {
+		return "Vertical Position"
+	} else if *pos == 7 {
+		return "Horizontal Position"
+	}
```

Comment:
@lmtan91 just to be clear, we don't want any code in here related to the FGA OEM product. This Rubix-Data-Encoding is meant to be generic for any device that implements it. Don't forget that the Dorma uart-to-lora oem product also uses this encoding format so their point IDs will mean something completely different. Therefore it is up to the user to name these points correctly from the UI (CE). This block should be removed